plugin.video.themoviedb.helper/resources/lib/requestapi.py

```diff
@@ -0,0 +1,148 @@
+import time
+import datetime
+import simplecache
+import xbmcaddon
+import xbmcgui
+import xbmc
+import requests
+import resources.lib.utils as utils
+import xml.etree.ElementTree as ET
+_cache = simplecache.SimpleCache()
+_xbmcdialog = xbmcgui.Dialog()
+_addon = xbmcaddon.Addon()
+
+
+class RequestAPI(object):
+    def __init__(self):
+        self.req_api_url = ''
+        self.req_api_key = ''
+        self.req_api_name = ''
+        self.req_wait_time = 1
+        self.cache_long = 14
+        self.cache_short = 1
+        self.addon_name = 'plugin.video.themoviedb.helper'
+
+    def invalid_apikey(self, api_name='TMDb'):
+        _xbmcdialog.ok('{0} {1} {2}'.format(_addon.getLocalizedString(32007), api_name, _addon.getLocalizedString(32008)),
+                       _addon.getLocalizedString(32009))
+        xbmc.executebuiltin('Addon.OpenSettings({0})'.format(self.addon_name))
```

Comment:
I'd prefer if you could specify the id of the addon you're trying to access the settings

Reply:
This was intentional so that a separate addon can use the plugin as a module to retrieve data from TMDb. In this scenario, the 2nd add-on would pass a tmdb apikey as an argument to the TMDb class (which inherits this RequestAPI class). In that case, my plugin's settings are no longer relevant as the 2nd add-on is specifying its own apikey in its own addon settings.